pkg/bundle/bundle.go

```diff
@@ -454,32 +453,13 @@ func (b *bundle) bundleTargetNamespaceSelector(bundleObj *trustapi.Bundle) (labe
 // to ensure that the apply operations will also remove fields that were
 // created by the Update operation.
 func (b *bundle) migrateBundleStatusToApply(ctx context.Context, obj client.Object) (bool, error) {
-	// isOldBundleStatusManagedFieldsEntry returns true if the given ManagedFieldsEntry is
-	// an entry that was created by the old fieldManager and is an update to the status
-	// subresource. We need to check for this as we need to migrate the entry to the new
-	// fieldManager.
-	isOldBundleStatusManagedFieldsEntry := func(mf *metav1.ManagedFieldsEntry) bool {
-		return (mf.Manager == fieldManager || mf.Manager == crRegressionFieldManager) &&
-			mf.Operation == metav1.ManagedFieldsOperationUpdate &&
-			mf.Subresource == "status"
-	}
-
-	needsUpdate := false
-	managedFields := obj.GetManagedFields()
-	for i, mf := range managedFields {
-		if !isOldBundleStatusManagedFieldsEntry(&mf) {
-			continue
-		}
-
-		needsUpdate = true
-		managedFields[i].Operation = metav1.ManagedFieldsOperationApply
-		managedFields[i].Manager = fieldManager
+	patch, err := csaupgrade.UpgradeManagedFieldsPatch(obj, sets.New(fieldManager, crRegressionFieldManager), fieldManager, csaupgrade.Subresource("status"))
+	if err != nil {
+		return false, err
 	}
-
-	if !needsUpdate {
-		return false, nil
+	if patch != nil {
+		return true, b.client.Patch(ctx, obj, client.RawPatch(types.JSONPatchType, patch))
```

Comment:
This is quite a neat trick: instead of update, patch (JSON instead of ssa) is used so there is no extra rbac necessary.